compiler/semantic/expr.go

```diff
@@ -992,6 +994,28 @@ func (a *analyzer) semFString(f *ast.FString) dag.Expr {
 	return out
 }
 
+func (a *analyzer) semQueryExpr(q *ast.QueryExpr) dag.Expr {
+	e := &dag.QueryExpr{
+		Kind: "QueryExpr",
+		Body: a.semSeq(q.Body),
+	}
+	// XXX We should probably add a check to return error on nonsensical
+	// subqueries like (head 1)?
```

Comment:
Is `head 1` really nonsensical since it will produce output (from the dag.NullScan added below)?

Reply:
I believe any subquery without a source (including values) is non-sensical

Reply:
We'll just let this be for now.